library/src/scala/util/boundary.scala

```diff
@@ -0,0 +1,40 @@
+package scala.util
+import control.ControlException
+
+/** A boundary that can be exited by `break` calls.
+ *  `boundary` and `break` represent a unified and superior alternative for the
+ *  `scala.util.control.NonLocalReturns` and `scala.util.control.Breaks` APIs.
+ *  The main differences are:
+ *
+ *    - Unified names: `boundary` to establish a scope, `break` to leave it.
+ *      `break` can optionally return a value.
+ *    - Integration with exceptions. `break`s are logically non-fatal exceptions.
+ *      The `Break` exception class extends `ControlException` which is a regular
+ *      `RuntimeException`, optimized so that stack trace generation is suppressed.
+ *    - Better performance: breaks to enclosing scopes in the same method can
+ *      be rwritten to jumps.
+ */
+object boundary:
```

Comment:
This will require a SIP. It should be marked as `@experimental` if we want to have it in `3.3.0`.

Reply:
It's a library and it does not interact with the language spec (like `Seq` does, for example), so it is outside the scope of SIPs. (the fact that the compiler chooses to *optimize* that particular piece of library is an implementation detail; not SIP material)